napari/_qt/qt_main_window.py

```diff
@@ -807,14 +807,21 @@ def _status_bar(self):
         # TODO: remove from window
         return self._qt_window.statusBar()
 
-    def _update_menu_state(self, menu):
-        """Update enabled/visible state of menu item with context."""
+    def _update_file_menu_state(self):
         layerlist = self._qt_viewer._layers.model().sourceModel()._root
-        menu_model = getattr(self, menu)
-        menu_model.update_from_context(get_context(layerlist))
+        self.file_menu.update_from_context(get_context(layerlist))
 
-    def _update_plugin_menu_state(self):
-        self._update_menu_state('plugins_menu')
```

Comment:
Actually previously we were defining update function, not using partial. Did we think partial was problematic?

I think @Czaki said string is fine as long as we type via `Literal`.

So partial for each mene or defining a method for each update function? @Czaki ?

Reply:
> I think @Czaki said string is fine as long as we type via `Literal`.

Yes. Because most often we use in codebase:

```
def fun(label: str):
    if label == "aa":
        # something
    elif label == "ab":
        # something
    elif label == "ac":
        # something
    else:
        # something
```
When allowed values are `"aa"`, `"ab"`, `"ac"`, `"ad"`. But the above code will also work for `"ae"`. There will be no error for such typo. Using literal allow to detect such error using mypy at least for part of cases (also IDE my highlight).

Another option is to use Enum to enforce consistency. StrEnum may allow having runtime exception for inproper values.  

> So partial for each mene or defining a method for each update function? @Czaki ?

Separate method. Partial connected to signal will keep hard reference to the object and keep it alive when it should be deleted.